typescript/agentkit/src/action-providers/opensea/openSeaActionProvider.ts

```diff
@@ -0,0 +1,141 @@
+import { z } from "zod";
+import { ActionProvider } from "../actionProvider";
+import { CreateAction } from "../actionDecorator";
+import { OpenSeaSDK, Chain } from "opensea-js"; // Assuming an OpenSea SDK is available
+import { Network } from "../../network";
+import { OpenSeaGetNftsByAccount, OpenSeaListNFTSchema } from "./schemas";
+import { ViemWalletProvider } from "../../wallet-providers";
+import { privateKeyToAccount } from "viem/accounts";
+import { createWalletClient, http } from "viem";
+import { sepolia } from "viem/chains";
+
+/**
+ * Configuration options for the OpenSeaActionProvider.
+ */
+export interface OpenSeaActionProviderConfig {
+  apiKey?: string;
+  walletPrivateKey: string;
+}
+
+/**
+ * OpenSeaActionProvider is an action provider for interacting with OpenSea.
+ *
+ * @augments ActionProvider
+ */
+export class OpenSeaActionProvider extends ActionProvider {
+  private readonly client: OpenSeaSDK;
+  private readonly walletProvider: ViemWalletProvider;
```

Comment:
I suggest extending `ViemWalletProvider` which will configure a PublicClient / WalletClient. Then you'll also be able to remove the `walletPrivateKey` config parameter